extensions/objects/templates/image-editor.component.ts

```diff
@@ -0,0 +1,1155 @@
+    const cy2 = this.cropArea.y2;
+
+    const xOnLeftBorder = x > cx1 - margin && x < cx1 + margin;
+    const xOnRightBorder = x > cx2 - margin && x < cx2 + margin;
+    const yOnTopBorder = y > cy1 - margin && y < cy1 + margin;
+    const yOnBottomBorder = y > cy2 - margin && y < cy2 + margin;
+    const xInside = x > cx1 && x < cx2;
+    const yInside = y > cy1 && y < cy2;
+
+    // It is important to check the pointer position for corners first,
+    // since the conditions overlap. In other words, the pointer can be
+    // at the top border and at the top-right corner at the same time, in
+    // which case we want to recognize the corner.
+    if (xOnLeftBorder && yOnTopBorder) {
+      // Upper left corner.
+      this.mousePositionWithinCropArea = this.MOUSE_TOP_LEFT;
+    } else if (xOnRightBorder && yOnTopBorder) {
+      // Upper right corner.
+      this.mousePositionWithinCropArea = this.MOUSE_TOP_RIGHT;
+    } else if (xOnLeftBorder && yOnBottomBorder) {
+      // Lower left corner.
+      this.mousePositionWithinCropArea = this.MOUSE_BOTTOM_LEFT;
+    } else if (xOnRightBorder && yOnBottomBorder) {
+      // Lower right corner.
+      this.mousePositionWithinCropArea = this.MOUSE_BOTTOM_RIGHT;
+    } else if (yOnTopBorder) {
+      // Top border.
+      this.mousePositionWithinCropArea = this.MOUSE_TOP;
+    } else if (xOnLeftBorder) {
+      // Left border.
+      this.mousePositionWithinCropArea = this.MOUSE_LEFT;
+    } else if (xOnRightBorder) {
+      // Right border.
+      this.mousePositionWithinCropArea = this.MOUSE_RIGHT;
+    } else if (yOnBottomBorder) {
+      // Bottom border.
+      this.mousePositionWithinCropArea = this.MOUSE_BOTTOM;
+    } else if (xInside && yInside) {
+      // Inside the crop area.
+      this.mousePositionWithinCropArea = this.MOUSE_INSIDE;
+    } else {
+      this.mousePositionWithinCropArea = null;
+    }
+  }
+
+  private getTrustedResourceUrlForImageFileName(imageFileName) {
+    if (
+      this.contextService.getImageSaveDestination() ===
+      AppConstants.IMAGE_SAVE_DESTINATION_LOCAL_STORAGE &&
+      this.imageLocalStorageService.isInStorage(imageFileName)) {
+      const imageUrl = this.imageLocalStorageService.getRawImageData(
+        imageFileName);
+      if (imageFileName.endsWith('.svg')) {
+        return this.svgSanitizerService.getTrustedSvgResourceUrl(imageUrl);
+      }
+      return imageUrl;
+    }
+    const encodedFilepath = window.encodeURIComponent(imageFileName);
+    return this.assetsBackendApiService.getImageUrlForPreview(
+      this.contextService.getEntityType(),
+      this.contextService.getEntityId(),
+      encodedFilepath);
+  }
+
+  resetFilePathEditor(): void {
+    if (
+      this.data.metadata.savedImageFilename && (
+        this.contextService.getImageSaveDestination() ===
+        AppConstants.IMAGE_SAVE_DESTINATION_LOCAL_STORAGE) &&
+      this.imageLocalStorageService.isInStorage(
+        this.data.metadata.savedImageFilename)
```

Comment:
Can we separate this condition into a function to make the code more readable?

Reply:
Seems out of scope of this PR, will do separately.